x/staking/keeper/delegation.go

```diff
@@ -99,6 +99,7 @@ func (k Keeper) RemoveDelegation(ctx sdk.Context, delegation types.Delegation) {
 	k.BeforeDelegationRemoved(ctx, delegation.DelegatorAddress, delegation.ValidatorAddress)
 	store := ctx.KVStore(k.storeKey)
 	store.Delete(types.GetDelegationKey(delegation.DelegatorAddress, delegation.ValidatorAddress))
+	k.ExportZeroDelegationForAccount(ctx, delegation.DelegatorAddress, delegation.ValidatorAddress)
```

Comment:
Should'nt this be after call of `ExportDelegationsForAccount`

Reply:
It shouldn't matter tbh; the two are mutually exclusive. The zero delegation is added because it explicitly does not exist in the AllDelegations* export.